client/main.lua

```diff
@@ -87,24 +87,56 @@ CreateThread(function ()
 end)
 
 local pedSpawned = false
-local peds = {basic = {}, adv ={}}
 local blips = {}
 function CreatePeds()
     if pedSpawned then return end
-    for k=1, #Config.peds do
-        local model = joaat(Config.peds[k].model)
-
-        RequestModel(model)
-        while not HasModelLoaded(model) do Wait(0) end
-
+    for k = 1, #Config.peds do
         local coords = Config.peds[k].coords
-        local bankPed = CreatePed(0, model, coords.x, coords.y, coords.z-1, coords.w, false, false)
+        local pedPoint = lib.points.new({
+            coords = coords,
+            distance = 300,
+            model = joaat(Config.peds[k].model),
+            ped = nil,
+            targetOptions = Config.peds[k].createAccounts and {{
+                name = 'renewed_banking_accountmng',
+                event = 'Renewed-Banking:client:accountManagmentMenu',
+                icon = 'fas fa-money-check',
+                label = locale('manage_bank'),
+                atm = false,
+                canInteract = function(_, distance)
+                    return distance < 4.5
+                end
+            }} or {{
+                name = 'renewed_banking_openui',
+                event = 'Renewed-Banking:client:openBankUI',
+                icon = 'fas fa-money-check',
+                label = locale('view_bank'),
+                atm = false,
+                canInteract = function(_, distance)
+                    return distance < 4.5
+                end
+            }}
+        })
+
+        function pedPoint:onEnter()
+            lib.requestModel(self.model, 10000)
+
+            self.ped = CreatePed(0, self.model, self.coords.x, self.coords.y, self.coords.z-1, self.coords.w, false, false)
+
+            TaskStartScenarioInPlace(self.ped, 'PROP_HUMAN_STAND_IMPATIENT', 0, true)
+            FreezeEntityPosition(self.ped, true)
+            SetEntityInvincible(self.ped, true)
+            SetBlockingOfNonTemporaryEvents(self.ped, true)
+            exports.ox_target:addLocalEntity(self.ped, self.targetOptions)
+        end
```

Comment:
You never do SetModelAsNoLongerNeeded, please do that.

Reply:
fixed